primitive-types/src/lib.rs

```diff
@@ -219,3 +219,45 @@ impl<'a> TryFrom<&'a U512> for U256 {
 		Ok(U256(ret))
 	}
 }
+
+impl From<f64> for U256 {
+	/// Lossy saturating conversion from a `f64` to a `U256`.
+	///
+	/// The conversion follows roughly the same rules as converting `f64` to other
+	/// primitive integer types. Namely, the conversion of `value: f64` behaves as
+	/// follows:
+	/// - `NaN` => `0`
+	/// - `(-∞, 0]` => `0`
+	/// - `(0, u256::MAX]` => `value as u256`
+	/// - `(u256::MAX, +∞)` => `u256::MAX`
+	fn from(value: f64) -> U256 {
+		if value >= 1.0 {
+			let bits = value.to_bits();
+			// NOTE: Don't consider the sign or check that the subtraction will
+			//   underflow since we already checked that the value is greater
+			//   than 1.0.
+			let exponent = ((bits >> 52) & 0x7ff) - 1023;
+			let mantissa = (bits & 0x0f_ffff_ffff_ffff) | 0x10_0000_0000_0000;
```

Comment:
does this implementation assume little-endianness?

Reply:
This is independent of endianness since we are not indexing bytes here.